### DISCUSSION TITLE
Email signup: PKCE flow is not supported on signups with autoconfirm enabled

### DISCUSSION THREAD

**Discussion by mbj36** [supabase/supabase#15992] [Questions]
What is the point of `Confirm Email` toggle when it doesn;t work ?
![Screenshot 2023-07-24 at 7 47 19 PM](https://github.com/supabase/supabase/assets/9936881/9c1744aa-392b-47d1-95ac-abb84cbc48a1)

**Comment by prnews-io-tech** [comment]
You can just change the flow type to `implicit`

```ts
clientOptions: {
     auth: {
        flowType: 'implicit'
    }
},
```

**Reply by mbj36** [reply]
Can you point me to docs where it is mentioned ?

**Reply by leandergangso** [reply]
I also don't seem to find this in the documentation.

**Reply by prnews-io-tech** [reply]
Sorry, i gave my answer right before the vacation and completely forgot why it was exactly like this 🤦‍♂️

Actually, this method works fine to me. You need to call `supabase.auth.signUp` and follow the confirmation link in the email (and you will be logged in automatically). I'm using `nuxt` implementation and session handling works out of a box. But it shouldn't be that hard to handle it manually, you can check [nuxt implementation](https://github.com/nuxt-modules/supabase/blob/main/src/runtime/plugins/supabase.client.ts) out by yourself

**Reply by leandergangso** [reply]
I figured out a solution by reading the source code.
I needed the PKCE flow in my case, since i'm doing stuff server side.
Ended up just including the `code_challenge_method` and `code_challenge` in the json body of the request.
If those values are present in the request body it uses PKCE.

**Comment by J0** [comment]
Hey 

Thanks everyone for helping each other out. Love the community spirit - you should be able to do this now.

Let us know if there are still issues!
